### ISSUE TITLE
DataView inlineFilter: true and setFilter throws read-only error on fn.name setting

### ISSUE THREAD

**Issue by jmzagorski** [DimitarChristoff/slickgrid-es6#14]
I am running into a problem running this example with SlickGrid http://mleibman.github.io/SlickGrid/examples/example5-collapsing.html. If `inlineFilter: true` is set in the `DataView` ctor and `setFilter` is called then the `compileFilter` function runs and attempts to set the `displayName` and `name` of the annoymouns function. The error returned from the browser is 

```
Unhandled rejection TypeError: Cannot assign to read only property 'name' of function 'function anonymous(_items,_args
/*``*/) {
```

**Comment by DimitarChristoff** [comment]
yep. https://github.com/DimitarChristoff/slickgrid-es6/blob/master/src/slick.dataview.js#L669

I did not use compiled filters because of this. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name#Function_constructor_name

I guess it can be changed to return `function compiledFilter(){ fn.apply(this,arguments)` or to return `new Function(fn.toSource().replace('function ', 'function compiledFilter'))`

or it can be dropped as it won't really matter - it gets assigned to a variable anyway. 

I've put it in a branch and it seems to work fine. https://github.com/DimitarChristoff/slickgrid-es6/pull/15/files - feel free to review/test it and let me know, happy to merge

**Comment by jmzagorski** [comment]
thanks. As soon as I get a chance I will pull down the `inline-filter-dataview`  branch into my app and test it to make sure filtering is still working as it was before.

**Comment by DimitarChristoff** [comment]
To be honest, I see no reason not to merge and release this as it appears
to work fine. Will do shortly and push 2.0.4

-- 
Dimitar Christoff

"JavaScript is to JAVA what hamster is to ham"
@D_mitar - https://github.com/DimitarChristoff

**Comment by jmzagorski** [comment]
sounds good to me

**Comment by DimitarChristoff** [comment]
closed with #14
